src/acceleration/AitkenAcceleration.cpp

```diff
@@ -55,28 +60,25 @@ void AitkenAcceleration::performAcceleration(
   // Compute aitken relaxation factor
   PRECICE_ASSERT(utils::contained(*_dataIDs.begin(), cplData));
 
-  Eigen::VectorXd values;
-  Eigen::VectorXd oldValues;
-  for (int id : _dataIDs) {
-    utils::append(values, cplData.at(id)->values());
-    utils::append(oldValues, Eigen::VectorXd(cplData.at(id)->previousIteration()));
-  }
+  concatenateCouplingData(cplData, _dataIDs, _values, _oldValues);
 
-  // Compute current residuals
-  Eigen::VectorXd residuals = values;
-  residuals -= oldValues;
+  // Compute current residual = values - oldValues
+  auto residuals = _values - _oldValues;
```

Comment:
Short info as this isn't obvious: [Eigen strongly discourages to use auto with their expressions](https://eigen.tuxfamily.org/dox/TopicPitfalls.html).

```suggestion
  Eigen::VectorXd residuals = _values - _oldValues;
```

Reply:
I was not aware of that, good point, thanks!